pyarr/base.py

```diff
@@ -47,13 +60,65 @@ def assert_return(
         assert isinstance(response, typearg)
         return response
 
+    def assert_return_put(
```

Comment:
I first tried updating the _put in the request_handler, but this can return either a list or a dict, I could change the return type to `Any` here which would resolve that issue, I'm not sure what is best to do here. 

Although I have the response from `assert_return_put` as `Any` which I guess is the same and moving it to `_put` removes that extra method.

Would like to know your thoughts on changing `_put`  return type to `Any` and then removing the `assert_return_put` methods?

Reply:
Yeah let's handle anything we need directly within `_put`.

Does maintaining the current `Union[dict[str, Any], list[dict[str, Any]]]` on `_put` cause type hint issues?

Reply:
Yes, if you leave the current type it then causes the arr command return type to error as its not an expected return type. This is caused because some commands return a `list` and others a `dict`

Reply:
Yeah let's just shove type validation logic into the base methods.

For example:

```python
_ReturnType = TypeVar("_ReturnType", bound=Union[dict, list])

def _put(... , typevar: _ReturnType, ...) -> _ReturnType:
    ...
```